### PR TITLE
[dag] broadcast nodes within window till all validators ack

### PR DESCRIPTION
**Pull request by ibalajiarun**

### CODE REVIEW

**Comment by ibalajiarun** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11751?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #11362** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11362?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #11407** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11407?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #11505** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11505?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #11685** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11685?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #11743** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11743?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #11751** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11751?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
        * **PR #11433** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11433?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11751?utm_source=stack-comment).

File: consensus/src/dag/dag_driver.rs [hunk 1]
```diff
@@ -93,7 +97,7 @@ impl DagDriver {
             payload_client,
             reliable_broadcast,
             time_service,
-            rb_abort_handle: Mutex::new(None),
+            rb_handles: Mutex::new(BoundedVecDeque::new(window_size_config as usize)),
```
Comment (sasha8) [inline consensus/src/dag/dag_driver.rs:100]: Do we have to use the same window size here?

**Review by sasha8 — APPROVED** [review]

File: consensus/src/dag/dag_driver.rs [hunk 2]
```diff
@@ -356,10 +360,37 @@ impl RpcHandler for DagDriver {
     }
 }
 
-impl Drop for DagDriver {
-    fn drop(&mut self) {
-        if let Some((handle, _)) = self.rb_abort_handle.lock().as_ref() {
-            handle.abort()
+struct BoundedVecDeque<T> {
```
Comment (zekun000) [inline consensus/src/dag/dag_driver.rs:363]: nit: probably move it to a util, and use it for the sliding window inside MetadataBackendAdapter too. also small unit tests would be good

**Review by zekun000 — APPROVED** [review]

File: consensus/src/dag/dag_driver.rs (hunk 1 above)
Comment (ibalajiarun) [reply consensus/src/dag/dag_driver.rs:100]: We could increase, but let's keep this for now.

**Pull request merged**